### DISCUSSION TITLE
recorded_values: exists a limit for output

### DISCUSSION THREAD

**Discussion by RoseDeSable** [Hugovdberg/PIconnect#739] [Q&A]
Hi,
 I'm reading out the values for AIPoints in intervals of 1 hour. In some outputs I see "..." within the sequence of the output lines. Exists a limit for lines ? If it is so, can i set it higher ? For example for numpy arrays such limits are set. The programmer can set it higher up to the by the system allowed maximum.

Best Regards
Rose

**Comment by Hugovdberg** [comment]
I think you are referring to the number of rows shown in a jupyter notebook? The data is a `pandas.Series` with some extra info, and therefore the number of displayed rows is controlled by pandas. See this page op the pandas documentation: https://pandas.pydata.org/docs/user_guide/options.html, I think you need `pd.options.max_rows`

**Reply by RoseDeSable** [reply]
Ok,
 I use this code:

import PIconnect as PI
with PI.PIServer() as server:
 points = server.search('*')
 data = points[1].recorded_values('*-1h', '*')
 print(data)

Sometimes the output is 

2024-03-12 14:16:31.301000+00:00    7097
2024-03-12 14:16:46.301000+00:00    7097
2024-03-12 14:17:38.302000+00:00    7097
2024-03-12 14:17:54.302000+00:00    7098
2024-03-12 14:18:46.303000+00:00    7098
                                    ... 
2024-03-13 13:32:51.400000+00:00    7892
2024-03-13 13:33:06.900000+00:00    7893
2024-03-13 13:33:56.901000+00:00    7893
2024-03-13 13:35:03.403000+00:00    7893
2024-03-13 13:35:18.903000+00:00    7894
Name: AF.myEnterprise.My.Ident-Nr, Length: 993, dtype: int64

Here you see the point line, that I mean. 

If I use  print(type(data)), I see the following

<class 'PIconnect.PIData.PISeries'>

But I see no documention of this class. I would be useful for me to know it.

**Reply by RoseDeSable** [reply]
Hello,
 I solved my problem with this coding:

with PI.PIServer() as server:
 points = server.search('*')
 data = points[1].recorded_values('*-48h', '*')
 for i in range(len(data.index)):
  print(data.name+" "+str(data.index[int(i)])+" "+str(data.array[int(i)]))

I can output all the data in lines.

Thanks
Rose
